application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -280,6 +318,69 @@ private boolean isWrongTagStatusAndHandle(@NotNull TagStatus requiredTagStatus,
         return false;
     }
 
+    private void logAction(@NotNull Subcommand subcommand, @NotNull Guild guild,
+            @NotNull User author, @NotNull TemporalAccessor timestamp, @NotNull String id,
+            @Nullable String newContentNullable, @Nullable String previousContentNullable) {
+
+        Optional<String> newContent =
+                (newContentNullable == null) ? Optional.empty() : Optional.of(newContentNullable);
+        Optional<String> previousContent = (previousContentNullable == null) ? Optional.empty()
+                : Optional.of(previousContentNullable);
```

Comment:
This is abuse of `Optional`. Do not use optional just for the sake of its nice api. Use null instead, no optionals.

Optional is only to be used as method return type.

Reply:
I am now checking for `null`, not sure if its better tho. please review https://github.com/Together-Java/TJ-Bot/commit/46f92eef616b6f7f263d8e256f5c8efd3095c347.